R/importSNV.TCGA.R

```diff
@@ -0,0 +1,268 @@
+        rowsBefore <- length(rSNVf)
+        myNotes <- c(myNotes, paste("Number of rows originally in rSNV - ", rowsBefore))
+        snvValidity <- .rSNVheadercheck(rSNV)
+    }
+
+    # ==== READ DATA ===========================================================
+    filesFinished <- 0
+    numFilesInvalid <- 0
+
+    for (mafFile in fMAF) {
+        if (length(.fileValidity(mafFile)) + length(.mafHeaderCheck(mafFile)) + length(snvValidity) == 0) {
+
+            fileasDT <- read.table(mafFile, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                        na.strings = " ")[,c(1,5,6,7,8,9,10,11,12,13,16,33)]
+
+            # check if variant_class contains valid values from known list
+
+            cat(sprintf("Processing file - \"%s\" - (%i/%i)", mafFile, filesFinished, length(fMAF)))
+
+            classValid <- TRUE
+            typeValid <- TRUE
+            chrValid <- TRUE
+
+            classVknown <- c("Frame_Shift_Del", "Frame_Shift_Ins", "In_Frame_Del", "In_Frame_Ins", "Missense_Mutation",
+                        "Nonsense_Mutation", "Silent", "Splice_Site", "Translation_Start_Site", "Nonstop_Mutation",
+                        "3'UTR", "3'Flank", "5'UTR", "5'Flank", "IGR1" , "Intron", "RNA", "Targeted_Region")
+            myVclasses <- unique(fileasDT[,6])
+
+            if (! all(myVclasses %in% classVknown)) {
+                # drop file
+                classValid <- FALSE
+                myNotes <- c(myNotes,paste(mafFile, " - was skipped because of unrecognized string in variant class column."))
+            }
+
+            # check if vairant_type contains valid values from known list
+            typeVknown <- c("SNP", "DNP", "TNP", "ONP", "INS", "DEL", "Consolidated2")
+            myVtypes <- unique(fileasDT[,7])
+            if (! all(myVtypes %in% typeVknown)) {
+                # drop file
+                typeValid <- FALSE
+                myNotes <- c(myNotes, paste(mafFile, " - was skipped because of unrecognized string in variant type column."))
+            }
+
+            # check chromosome value is valid
+            typeChr <- c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13",
+                         "chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chrX","chrY","chrM")
+            myVchr <- unique(fileasDT[,2])
+            if (! all(myVchr %in% typeChr)) {
+                # drop file
+                chrValid <- FALSE
+                myNotes <- c(myNotes, paste(mafFile, " - was skipped because of unrecognized string in variant chromosome column."))
+            }
+
+            if (! classValid && typeValid && chrValid) {
+                # there was an invalid value
+                # drop file, go to next
+                numFilesInvalid <- numFilesInvalid + 1
+            } else {
+                # all tests passed
+                # add data to rSNV
+                if (na.rm) {
```

Comment:
Consider this bit of code (essentially from the examples of ```na.omit()```):
```
(myDF <- data.frame(x = c(1, 2, 3), y = c(0, 10, NA)))
na.omit(myDF[, 1])
na.omit(myDF[, 2])
```
... I don't think the result of your expression is what you intend it to be.